functions/Get-DbaAgentAlert.ps1

```diff
@@ -71,17 +84,26 @@ function Get-DbaAgentAlert {
 
             $alerts = $server.Jobserver.Alerts
 
-            foreach ($alert in $alerts) {
-                $lastraised = [dbadatetime]$alert.LastOccurrenceDate
+            if (Test-Bound 'Alert') {
+                $alerts = $alerts | where Name -like $Alert;
+            }
+
+
+            if (Test-Bound 'ExcludeAlert') {
+                $alerts = $alerts | where Name -notlike $ExcludeAlert;
```

Comment:
It should handle both scenarios where they provided a wildcard or they only want one to return. Current state if I provided "Backup" it will return any job that has the term "backup" in the job name and not just the one I called "Backup".

Reply:
We have another command that handles both scenarios in the same parameter, but I'm forgetting which one it is.

Reply:
I don't want to break functionality of the current parameter. The other commands I think we just check for the `*`... `if ($Alert -match '`*') {`, or something of that nature.

Reply:
Hmm... I tested it just now and it doesn't seem to work that way. On my local instance, I have several alerts that replication installed. If I run `Get-DbaAgentAlert -SqlInstance . -ExcludeAlert 'Replication'` I get all of the alerts (i.e. no filtering happened as none of the alerts are called `Replication`). Contrast that with if I run `Get-DbaAgentAlert -SqlInstance . -ExcludeAlert 'Replication*'` then those alerts are excluded as expected.

I'd expect it to work as you describe if the `-notmatch` operator were used instead of `-notlike`.